src/NLog/Fluent/LogBuilder.cs

```diff
@@ -367,38 +363,35 @@ public void WriteIf(Func<bool> condition)
         /// <param name="callerFilePath">The full path of the source file that contains the caller. This is set at by the compiler.</param>
         /// <param name="callerLineNumber">The line number in the source file at which the method is called. This is set at by the compiler.</param>
         public void WriteIf(
-            bool condition, 
-            [CallerMemberName]string callerMemberName = null, 
-            [CallerFilePath]string callerFilePath = null, 
-            [CallerLineNumber]int callerLineNumber = 0)
+            bool condition,
+#if NET4_5
+            [CallerMemberName]
+#endif
+            string callerMemberName = null,
+#if NET4_5
+            [CallerFilePath]
+#endif
+            string callerFilePath = null,
+#if NET4_5
+            [CallerLineNumber]
+#endif
+            int callerLineNumber = 0)
         {
             if (!condition || !_logger.IsEnabled(_logEvent.Level))
                 return;
 
+            // TODO NLog 5.0 Remove these properties
```

Comment:
this is for NLog 5? So we could remove this?

Reply:
Well it is todo work. But I would like to do it in separate commit. Like to keep things apart, since removing these properties requires cleanup other places.

Reply:
OK 👍